### ISSUE TITLE
CI performance regression and time-outs 

### ISSUE THREAD

**Issue by AaronFriel** [pulumi/pulumi#11022]
Multiple CI jobs timed out while trying to release v3.43.0.

Investigate if this is a performance regression (kind/bug) or external issue with GitHub runners.

**Comment by AaronFriel** [comment]
In the most recent successful job, the `codegen/go` package failed on its first attempt before automatically being retried: https://github.com/pulumi/pulumi/actions/runs/3246233657/jobs/5324770211

No error message was logged.

**Comment by AaronFriel** [comment]
A prior run show some unusually long execution times for some codegen tests:

```
TestGenerateProgram/invalid-go-sprintf	github.com/pulumi/pulumi/pkg/v3/codegen/go	9m22.13s	pass
TestGenerateProgram/kubernetes-template	github.com/pulumi/pulumi/pkg/v3/codegen/go	8m46.27s	pass
TestGenerateProgram/kubernetes-operator	github.com/pulumi/pulumi/pkg/v3/codegen/go	8m26.37s	pass
TestGeneratePackage/external-go-import-aliases	github.com/pulumi/pulumi/pkg/v3/codegen/go	6m52.27s	pass
```

**Comment by iwahbe** [comment]
Another example: https://github.com/pulumi/pulumi/actions/runs/3268337965/jobs/5374657651

**Comment by AaronFriel** [comment]
Resolved by #11052
